scripts/tests/test_zkai_attention_kv_d32_four_head_seq32_fused_softmax_table_native_gate.py

```
@@ -0,0 +1,380 @@
+            self.assertEqual(json.loads(json_path.read_text(encoding="utf-8"))["decision"], gate.DECISION)
+            self.assertIn(gate.DECISION, tsv_path.read_text(encoding="utf-8"))
+            self.assertIn("154670", tsv_path.read_text(encoding="utf-8"))
+
+    def test_input_helpers_reject_symlink_paths(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_path = gate.pathlib.Path(tmp_dir)
+            script_target = tmp_path / "script_target.py"
+            script_target.write_text("VALUE = 1\n", encoding="utf-8")
+            script_link = tmp_path / "script_link.py"
+            script_link.symlink_to(script_target)
+            with self.assertRaisesRegex(gate.AttentionKvD32FourHeadSeq32FusedSoftmaxTableGateError, "symlink"):
```

Ruff (0.15.13)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)